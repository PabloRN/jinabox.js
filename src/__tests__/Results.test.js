import Results from '../Results';
import '../../golbalTestsSetup';
import { getByTestId, } from '@testing-library/dom'

describe('jina-results tests', () => {
    let container;

    beforeAll(() => {
        window.customElements.define('jina-results', Results);
    });

    beforeEach(() => {
        const div = document.createElement('div');
        div.innerHTML = `<jina-results></jina-results>`;
        container = div;
    });

    it('should get default inner HTML when first render', async () => {
        const jinaResults = getByTestId(container, 'jina-results-area');
        expect(jinaResults).toContainHTML(`<div class="jina-results-area" id="jina-results-area"></div>`);
        expect(container).toMatchSnapshot();
    });
});
