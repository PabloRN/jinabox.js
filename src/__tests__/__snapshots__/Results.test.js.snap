// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jina-results tests should get default inner HTML when first render 1`] = `
<div>
  Results {
    "dropArea": null,
    "handleDrag": [Function],
    "handleDragLeave": [Function],
  }
</div>
`;
